library/ColorText.cpp

```diff
@@ -196,6 +196,7 @@ void color_ostream_proxy::flush_proxy()
             target->add_text(it->first, it->second);
 
         target->end_batch();
+        target->flush_proxy();
```

Comment:
What if the buffer for this proxy is empty? Do we still need to flush the target?

Reply:
if the buffer is empty, then nothing has been written to the proxy since it was last flushed, and since the last flush would have caused the target to be flushed, there should be no need for another flush